packages/openactive-integration-tests/test/features/approval/minimal-proposal/implemented/accept-proposal-book-test.js

```diff
@@ -31,101 +30,68 @@ FeatureHelper.describeFeature(module, {
   // even if some OrderItems don't require approval, the whole Order should
   controlOpportunityCriteria: 'TestOpportunityBookable',
 },
-(configuration, orderItemCriteria, featureIsImplemented, logger, state, flow) => {
-  beforeAll(async () => {
-    await state.fetchOpportunities(orderItemCriteria);
+(configuration, orderItemCriteria, featureIsImplemented, logger) => {
+  // ## Initiate Flow Stages
+  const fetchOpportunities = FetchOpportunitiesFlowStage.create({ orderItemCriteria, logger });
+  const c1 = C1FlowStage.create({ preRequisite: fetchOpportunities, logger });
+  const c2 = C2FlowStage.create({ preRequisite: c1, logger });
+  const p = PFlowStage.create({ preRequisite: c2, logger });
+  const simulateSellerApproval = TestInterfaceActionFlowStage.create({
+    testName: 'Simulate Seller Approval (Test Interface Action)',
+    preRequisite: p,
+    createActionFn: () => ({
+      type: 'test:SellerAcceptOrderProposalSimulateAction',
+      objectType: 'OrderProposal',
+      objectId: p.getResult().body['@id'],
+    }),
+    logger,
   });
-
-  describe('Get Opportunity Feed Items', () => {
-    (new GetMatch({
-      state, flow, logger, orderItemCriteria,
-    }))
-      .beforeSetup()
-      .successChecks()
-      .validationTests();
-  });
-
-  describe('C1', () => {
-    (new C1({
-      state, flow, logger,
-    }))
-      .beforeSetup()
-      .successChecks()
-      .validationTests();
-
-    itShouldReturnOrderRequiresApprovalTrue(() => state.c1Response);
-  });
-
-  describe('C2', () => {
-    (new C2({
-      state, flow, logger,
-    }))
-      .beforeSetup()
-      .successChecks()
-      .validationTests();
-
-    itShouldReturnOrderRequiresApprovalTrue(() => state.c2Response);
+  const orderFeedUpdate = OrderFeedUpdateFlowStage.create({
+    testName: 'Order Feed (after Simulate Seller Approval)',
+    // orderFeedUpdate must be initiated by simulateSellerApproval because the
+    // order may appear on the feed as soon as simulateSellerApproval is
+    // completed.
+    initiatedBy: simulateSellerApproval,
```

Comment:
Great minds again! I totally agree. This is mostly a placeholder for now. I don't like that this approach means that some secret background mutation needs to occur, which might be surprising, but I do like that it means that the stages are written in order. The solution to [your other comment about using an array](https://github.com/openactive/openactive-test-suite/pull/215#discussion_r502764202) might have a way to have it both ways.

But I agree that in the less simple case, it should be the least magic version

Reply:
@nickevansuk in the end I split it into 2 FlowStages, which is very non-magic (quite boilerplate even!) - an "initiator" stage and a "collector" stage. This means that the core FlowStage logic doesn't need special affordances for this kind of use case (initiating feed updates before the prerequisite stage).

However, my plan is to have it so that the magic function on top will abstract away these details so that you don't have to think about creating 2 FlowStages.

Therefore, it's rather boilerplate-y now, but that's just the "low-level" code, which will generally be abstracted away. Lmkwyt